service/core/kafka/server/src/main/scala/com/lightbend/lagom/internal/broker/kafka/TopicProducerActor.scala

```diff
@@ -103,23 +102,36 @@ private[lagom] class TopicProducerActor[Message](
 
   def receive: Receive = {
     case Start => {
-      val backoffSource: Source[Future[Done], NotUsed] = {
+      val backoffSource: Source[Future[AkkaOffset], NotUsed] = {
```

Comment:
The stream no longer ends on `Future[Done]`. Instead, we propagate the `Offset` which can be used to identify what element just passed across the stream.